x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -826,6 +860,36 @@ export class TimeSeriesExplorer extends React.Component {
     if (detectorId !== selectedDetectorIndex) {
       appStateHandler(APP_STATE_ACTION.SET_DETECTOR_INDEX, detectorId);
     }
+    // if the detector's function is metric, fetch the highest scoring anomaly record
```

Comment:
I guess this logic can be moved outside of the TimeSeriesExplorer component

Reply:
Refactored here https://github.com/elastic/kibana/pull/81662/commits/19e20ee5f7a61327b729440a84a8fdf6e87a6a4a